examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,206 @@
+"""
+Title: Image Enhancer/Denoiser
+Author: [Anish B](https://twitter.com/anishhacko)
+Date created: 2021/12/31
+Last modified: 2022/01/05
+Description: Implementation of Image Enhancer/Denoiser Using U-Net.
+"""
+"""
+## Introduction
+
+This example signifies a "Image Enhancer/Denoiser" using UNet architecture with 
+pretrained ResNet50 as encoder and Sub-Pixel CNN as a decoder, we have followed up on 
+seminal papers as reference and used U-Net architecture with pretrained network(ResNet) 
+as encoder and Efficient Sub-Pixel CNN as decoder. 
+The main difference between Image Denoiser and Super-Resolution is Super-Resolution tries to 
+upscale the image than the size of its input dimension enhancing feature representation 
+but Image Denoiser tries to restore useful information with reasonable perceptual quality 
+eliminating the observed noises. The problem we focus here is denoising the document images 
+which are deteriorated through external degradations like blur,corrupt text blocks etc...
+
+**References:**
```

Comment:
Please follow style guidelines from https://github.com/keras-team/keras-io/blob/master/contributor_guide.md in particular with regard to list formatting, link formatting, etc.